runelite-client/src/main/java/net/runelite/client/plugins/combatlevel/CombatLevelOverlay.java

```diff
@@ -0,0 +1,169 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.combatlevel;
+
+import net.runelite.api.Client;
+import net.runelite.api.Skill;
+import net.runelite.api.VarClientInt;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.tooltip.Tooltip;
+import net.runelite.client.ui.overlay.tooltip.TooltipManager;
+import javax.inject.Inject;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import static java.lang.Math.floor;
+
+public class CombatLevelOverlay extends Overlay
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private CombatLevelConfig config;
+
+	@Inject
+	private TooltipManager tooltipManager;
+
+	private final Map<Skill, Integer> skillsLeft = new LinkedHashMap<>();
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showLevelsUntil() || client.getLocalPlayer().getCombatLevel() == 126
+			|| client.getVar(VarClientInt.CURRENT_TAB) != 0)
+		{
+			return null;
+		}
+
+		Widget combatLevelWidget = client.getWidget(WidgetInfo.COMBAT_LEVEL);
+		Rectangle combatCanvas = combatLevelWidget.getBounds();
+
+		if (combatCanvas.contains(client.getMouseCanvasPosition().getX(), client.getMouseCanvasPosition().getY()))
+		{
+			getLevelsUntil();
+			StringBuilder sb = new StringBuilder();
+			sb.append("<col=ff981f>Next combat level:</col></br>");
+
+			for (Map.Entry<Skill, Integer> entry : skillsLeft.entrySet())
+			{
+				String skillType;
+				switch (entry.getKey())
+				{
+					case ATTACK:
+						skillType = " Attack/Strength";
+						break;
+					case DEFENCE:
+						skillType = " Defence/Hitpoints";
+						break;
+					default:
+						skillType = " " + entry.getKey().getName();
+				}
+
+				sb.append(entry.getValue()).append(skillType).append("</br>");
+			}
+
+			tooltipManager.add(new Tooltip(sb.toString()));
+		}
+
+		return null;
+	}
+
+	private void getLevelsUntil()
```

Comment:
you could probably just merge this with render method/return string builder instead of using map and then iterating through, and at places where you put to map instead append to stringbuilder